### ISSUE TITLE
takeScreenshot API call with JSON output returns "false" instead of an ID

### ISSUE THREAD

**Issue by ebuchlin** [Helioviewer-Project/api#80]
A user has detected this on helioviewer-api.ias.u-psud.fr but api.helioviewer.org has the same issue: takeScreenshot with JSON output (`display=false`) systematically gives `{"id":false}`.

For example, it is the case with the example https://api.helioviewer.org/v2/takeScreenshot/?date=2014-01-01T23:59:59Z&imageScale=2.4204409&layers=[SDO,AIA,AIA,335,1,100]&events=[AR,HMI_HARP;SPoCA,1],[CH,all,1]&eventsLabels=false&x0=0&y0=0&width=1920&height=1200 from https://api.helioviewer.org/docs/v2/ .

Adding `&display=true` to the URL gives a PNG screenshot. This means that the data are present, and there should be nothing preventing a screenshot ID to be returned by takeScreenshot.

**Comment by dgarciabriseno** [comment]
I figured out this was caused by the screenshot table missing some columns.
I added the new columns to db.py in 182eda5

You can fix this with the following SQL command.
```sql
ALTER TABLE screenshots
ADD COLUMN celestialBodiesLabels VARCHAR(372),
ADD COLUMN celestialBodiesTrajectories VARCHAR(372)
```

**Comment by nimatraore** [comment]
Thank you! The new columns are now added to the database. However, it seems the output still always displays `{"id":false}`.

**Comment by dgarciabriseno** [comment]
In ImgIndex.php around line 105 in the function "insertScreenshot" there is a try/catch block that is hiding the error and returning false.
Add this echo (or use file_put_contents so you don't disrupt the service) so it logs what's breaking it.

![image](https://user-images.githubusercontent.com/94071409/148131487-9d76246c-d398-47dc-abd3-f45c9bb3672e.png)

**Comment by nimatraore** [comment]
Greayt, here is the log:

`Error executing database query (INSERT INTO screenshots SET id  = NULL, timestamp  = CURRENT_TIMESTAMP, observationDate  ='2014-01-01 23:59:59', imageScale  = 2.420441, regionOfInterest  = ST_PolygonFromText('POLYGON((-2323.623264 -1452.26454,2323.623264 -1452.26454,2323.623264 1452.26454,-2323.623264 1452.26454,-2323.623264 -1452.26454))'), watermark  = 1, dataSourceString  ='[SDO,AIA,AIA,335,1,100]', dataSourceBitMask  = 16384, eventSourceString  ='[AR,HMI_HARP;SPoCA,1],[CH,all,1]', eventsLabels  = 0, movieIcons  = 0, scale  = 0, scaleType  ='earth', scaleX  = 0.000000, scaleY  = 0.000000, numLayers  = 1, switchSources  = 0, celestialBodiesLabels = '', celestialBodiesTrajectories = '';): Unknown column 'switchSources' in 'field list'{"id":false}`

**Comment by ebuchlin** [comment]
@nimatraore is there a column `switchSources` in our `screenshots` table?
It is included in [db.py](https://github.com/Helioviewer-Project/api/blob/182eda5f737dc09675610548a93d8075428ad94c/install/helioviewer/db.py#L859).

**Comment by nimatraore** [comment]
Indeed, there is no `switchSources` in our `screenshots` table

**Comment by nimatraore** [comment]
Below the `screenshots` table:

![image](https://user-images.githubusercontent.com/96873544/148135749-687718fc-800b-4140-932d-201af04c97aa.png)

**Comment by nimatraore** [comment]
After adding `switchSources` in the table, it works fine !

**Comment by nimatraore** [comment]
Thank you very much!

**Comment by dgarciabriseno** [comment]
Fixed in 3.3.3
